heareval/tasks/pipeline.py

```diff
@@ -575,18 +689,17 @@ class FinalizeCorpus(WorkTask):
 
     def requires(self):
         # Will copy the resampled data and the traintestmeta and the vocabmeta
-        splits = [p["name"] for p in self.data_config["splits"]]
         return {
-            "resample": [
-                ResampleSubCorpus(
-                    sr=sr,
-                    split=split,
-                    metadata=self.metadata,
-                    data_config=self.data_config,
-                )
-                for sr in self.sample_rates
-                for split in splits
-            ],
+            "stats": GenerateAudioStats(
```

Comment:
We want to do this on the original folders before ANY preprocessing also. I don't think this should be a task. It should just be a function you call within tasks.

Reply:
Ya agreed - I think we really want the report on the duration percentiles *before* we do any preprocessing and as a function that we call so we can get insight into a dataset and decide on what duration to set for MonoWavTrimCorpus, etc.. -- so this will be mega helpful prior to building out the full pipeline. 

This may be helpful as a sanity check at the end of the pipeline. But we would expect all the durations to be the same at this point.

We did discuss that it would be nice to get a stats report out detailing the number of audio files in each split and the different labels and distribution at the end of a pipeline ya?

Reply:
Yes, We can remove it from being a task. We will have to track the different statistic JSON in different work folders though, to copy them in the final task(outside _workdir) output. In this task, we are just making one JSON, kind of the summary of what was the input and the output, and then copying this JSON over to the task dir(outside the `_workdir`). We can still go back to having this as a function. Please let me know your thoughts.

Reply:
But I get the requirement of setting the durations for the `MonoWavTrimCorpus`. We can make it a function and call in the extract metadata to save the stats in the same folder as the metadata. This is actually kind of a metadata of the input files in that case. We can keep the `GenerateAudioStats` in case we want to generate any other stats along with the once we are doing. Or we can remove as well.

Reply:
Ya -- I kind of like having this reported in the task dir as a summary of the input vs output -- but I think it's more useful from a debugging perspective, so maybe something we could optionally include as a task in the pipeline to help when building? @turian ?

A function that we can call on a folder of audio files that looks through recursively and reads the audio files and produces a report using the stats you have currently would be awesome.

Reply:
@khumairraj Again, we need the for datasets at the BEGINNING (before trim) and at the end. We want it at the beginning so we can actually pick appropriate audio file lengths for each particular task.